Set up

GeorgieGirl24 opened the pull request:
### Description

* This sets up the Flask BE app
* Models created include:
      * User
      * Quest
      * UserQuest
      * Encounter
      * Action
* Travis CI was successfully set to deploy to Heroku
* The `manage.py` file also has the seeds for the db

### Closes issue(s)

closes #2 
closes #3 
closes #11 

### Testing

* There are no additional test in this PR

### Screenshots

### Changes include
- [ ] Bugfix (non-breaking change that solves an issue)
- [ x] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (change that is not backwards-compatible and/or changes current functionality)

### Checklist
- [ ] I have written tests for this code (happy & sad)
- [ x] I have updated the Readme
- [ ] I ran full test suite - all tests passing

### Other comments
Run `dropdb database jobquest_dev;`
Run `dropdb jobquest_test;`
Run `createdb jobquest_dev;`
Run `createdb jobquest_test;`

Run `export DATABASE_URL=postgresql://localhost:5432/jobquest_dev`
Run `python manage.py db migrate`
Run `python manage.py db upgrade`

Run `export DATABASE_URL=postgresql://localhost:5432/jobquest_test`
Run `python manage.py db upgrade`

Run `python manage.py db_seed` on your local

GeorgieGirl24 committed:
Update travis.yml with correct config

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com> Co-authored-by: Jake Heft 
<60531761+jakeheft@users.noreply.github.com> Co-authored-by: Carson 
Jardine <65981543+carson-jardine@users.noreply.github.com> 
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

Co-Authored-By: Shaunda Cunningham <4582791+smcunning@users.noreply.github.com>
Co-Authored-By: Curtis Bartell <60277914+c-bartell@users.noreply.github.com>
Co-Authored-By: Carson Jardine <65981543+carson-jardine@users.noreply.github.com>
Co-Authored-By: Jake Heft <60531761+jakeheft@users.noreply.github.com>

GeorgieGirl24 committed:
Update the travis.yml

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

Co-Authored-By: Shaunda Cunningham <4582791+smcunning@users.noreply.github.com>
Co-Authored-By: Curtis Bartell <60277914+c-bartell@users.noreply.github.com>
Co-Authored-By: Carson Jardine <65981543+carson-jardine@users.noreply.github.com>
Co-Authored-By: Jake Heft <60531761+jakeheft@users.noreply.github.com>

GeorgieGirl24 committed:
Update travis.yml

Co-authored-by: Shaunda Cunningham <4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine <65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell <60277914+c-bartell@users.noreply.github.com>

GeorgieGirl24 committed:
Remove line in travis.yml

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

Co-Authored-By: Shaunda Cunningham <4582791+smcunning@users.noreply.github.com>
Co-Authored-By: Curtis Bartell <60277914+c-bartell@users.noreply.github.com>
Co-Authored-By: Carson Jardine <65981543+carson-jardine@users.noreply.github.com>
Co-Authored-By: Jake Heft <60531761+jakeheft@users.noreply.github.com>

GeorgieGirl24 committed:
Update .travis.yml with different app name 

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

Co-Authored-By: Shaunda Cunningham <4582791+smcunning@users.noreply.github.com>
Co-Authored-By: Curtis Bartell <60277914+c-bartell@users.noreply.github.com>
Co-Authored-By: Carson Jardine <65981543+carson-jardine@users.noreply.github.com>
Co-Authored-By: Jake Heft <60531761+jakeheft@users.noreply.github.com>

GeorgieGirl24 committed:
Add a uri to the config.py file

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

Co-Authored-By: Shaunda Cunningham <4582791+smcunning@users.noreply.github.com>
Co-Authored-By: Curtis Bartell <60277914+c-bartell@users.noreply.github.com>
Co-Authored-By: Carson Jardine <65981543+carson-jardine@users.noreply.github.com>
Co-Authored-By: Jake Heft <60531761+jakeheft@users.noreply.github.com>

GeorgieGirl24 committed:
Update the travis.yml file with additional line 
 
Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

Co-Authored-By: Shaunda Cunningham <4582791+smcunning@users.noreply.github.com>
Co-Authored-By: Curtis Bartell <60277914+c-bartell@users.noreply.github.com>
Co-Authored-By: Carson Jardine <65981543+carson-jardine@users.noreply.github.com>
Co-Authored-By: Jake Heft <60531761+jakeheft@users.noreply.github.com>

GeorgieGirl24 committed:
Add a 'ql' to our postgres call 

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

Co-Authored-By: Shaunda Cunningham <4582791+smcunning@users.noreply.github.com>
Co-Authored-By: Curtis Bartell <60277914+c-bartell@users.noreply.github.com>
Co-Authored-By: Carson Jardine <65981543+carson-jardine@users.noreply.github.com>
Co-Authored-By: Jake Heft <60531761+jakeheft@users.noreply.github.com>

GeorgieGirl24 committed:
Update travis.yml and config.py with enviornmental variables

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

Co-Authored-By: Shaunda Cunningham <4582791+smcunning@users.noreply.github.com>
Co-Authored-By: Curtis Bartell <60277914+c-bartell@users.noreply.github.com>
Co-Authored-By: Carson Jardine <65981543+carson-jardine@users.noreply.github.com>
Co-Authored-By: Jake Heft <60531761+jakeheft@users.noreply.github.com>

GeorgieGirl24 committed:
Update the submodels of database to handle models 

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

Co-Authored-By: Shaunda Cunningham <4582791+smcunning@users.noreply.github.com>
Co-Authored-By: Curtis Bartell <60277914+c-bartell@users.noreply.github.com>
Co-Authored-By: Carson Jardine <65981543+carson-jardine@users.noreply.github.com>
Co-Authored-By: Jake Heft <60531761+jakeheft@users.noreply.github.com>

GeorgieGirl24 committed:
Add models to the models.py

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

Co-Authored-By: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-Authored-By: Shaunda Cunningham <4582791+smcunning@users.noreply.github.com>
Co-Authored-By: Curtis Bartell <60277914+c-bartell@users.noreply.github.com>
Co-Authored-By: Carson Jardine <65981543+carson-jardine@users.noreply.github.com>
Co-Authored-By: Jake Heft <60531761+jakeheft@users.noreply.github.com>

GeorgieGirl24 committed:
Add migrations to the db

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

Co-Authored-By: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-Authored-By: Shaunda Cunningham <4582791+smcunning@users.noreply.github.com>
Co-Authored-By: Curtis Bartell <60277914+c-bartell@users.noreply.github.com>
Co-Authored-By: Carson Jardine <65981543+carson-jardine@users.noreply.github.com>
Co-Authored-By: Jake Heft <60531761+jakeheft@users.noreply.github.com>

GeorgieGirl24 committed:
Update encounters column to type instead of quest_type

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

Co-Authored-By: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-Authored-By: Shaunda Cunningham <4582791+smcunning@users.noreply.github.com>
Co-Authored-By: Curtis Bartell <60277914+c-bartell@users.noreply.github.com>
Co-Authored-By: Carson Jardine <65981543+carson-jardine@users.noreply.github.com>
Co-Authored-By: Jake Heft <60531761+jakeheft@users.noreply.github.com>

GeorgieGirl24 committed:
Add updated column to encounters table

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

Co-Authored-By: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-Authored-By: Shaunda Cunningham <4582791+smcunning@users.noreply.github.com>
Co-Authored-By: Curtis Bartell <60277914+c-bartell@users.noreply.github.com>
Co-Authored-By: Carson Jardine <65981543+carson-jardine@users.noreply.github.com>
Co-Authored-By: Jake Heft <60531761+jakeheft@users.noreply.github.com>

smcunning committed:
Create PR template

GeorgieGirl24 committed:
Update models with different configuration of the ForeignKey

GeorgieGirl24 committed:
Update migrations with chaning bugs fixes

GeorgieGirl24 committed:
Update seeds for the tables in the db

GeorgieGirl24 committed:
Remove unneed code

GeorgieGirl24 committed:
Merge branch 'set-up' of github.com:JobQuest/jobquest-be into set-up

GeorgieGirl24 committed:
Update tests to reflect the addition of timestamp

GeorgieGirl24 committed:
Remove whitespace

GeorgieGirl24 committed:
Update travis.yml

GeorgieGirl24 committed:
Update travis.yml

GeorgieGirl24 committed:
Update travis.yml

GeorgieGirl24 committed:
Update travis.yml

GeorgieGirl24 committed:
Update travis.yml

GeorgieGirl24 committed:
Update travis.yml

GeorgieGirl24 committed:
Update travis.yml

GeorgieGirl24 committed:
Update travis.yml file

GeorgieGirl24 committed:
Update travis.yml file

GeorgieGirl24 committed:
Update models to accurately reflect changes in the schema

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

GeorgieGirl24 committed:
Update migrates with seeded tables

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

GeorgieGirl24 committed:
Update tests to troubleshoot Travis by commenting out tests

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

GeorgieGirl24 committed:
Update postgres version to 9.6

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

GeorgieGirl24 committed:
Update tests with commented out tests to work on Travis 

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

GeorgieGirl24 committed:
Update test files with commented out code to address Travis issuses

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

GeorgieGirl24 committed:
Update the test files to address Travis issues

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

GeorgieGirl24 committed:
Update Travis and update the database in the config file to reflect test

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

GeorgieGirl24 committed:
Update the config file to not hard code the url

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

GeorgieGirl24 committed:
Deleted old migrations

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

GeorgieGirl24 committed:
Update the test file with one passing test

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

GeorgieGirl24 committed:
Update travis without an api key

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

GeorgieGirl24 committed:
Update travis.yml file

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

smcunning commented on api/database/submodels/quest.py:
```diff
@@ -0,0 +1,73 @@
+import bleach
```
Just a quick comment - since we're not using the submodels right now we shouldn't merge these into main.

GeorgieGirl24 replied:
You got it! Done and Done!!!

c-bartell reviewed (approved):
This looks great!

carson-jardine reviewed (approved):
LGTM!!!

GeorgieGirl24 committed:
Remove unneeded folder/files 

Co-authored-by: Shaunda Cunningham 
<4582791+smcunning@users.noreply.github.com>
Co-authored-by: Jake Heft <60531761+jakeheft@users.noreply.github.com>
Co-authored-by: Carson Jardine 
<65981543+carson-jardine@users.noreply.github.com>
Co-authored-by: Curtis Bartell 
<60277914+c-bartell@users.noreply.github.com>

jakeheft commented on api/database/submodels/quest.py:
```diff
@@ -0,0 +1,73 @@
+        the model must exist in the database
+        """
+        db.session.delete(self)
+        db.session.commit()
```
Don't need this file

jakeheft commented on manage.py:
```diff
@@ -24,12 +24,70 @@ def db_seed():
+    slay_dragon = Quest(name='Slay Argoroth the Dragon', xp=1000, encounter_req=5, type='active', level=3)
+    #
+    # # user_quests
+    user_quest_1 = UserQuest(quest_id=1, user_id=1, completion_status=False, progress=1)
+    user_quest_2 = UserQuest(quest_id=2, user_id=2, completion_status=False, progress=2)
+    user_quest_3 = UserQuest(quest_id=3, user_id=3, completion_status=False, progress=1)
+    #
```
We need to come back to these once we figure out to use the id of another variable.

jakeheft reviewed (approved):
Nice work getting this going!!
